src/vcpkg/commands.build.cpp

```diff
@@ -1488,54 +1488,75 @@ namespace vcpkg
-                if (first_block_end == std::string::npos || first_block_end > START_BLOCK_MAX_LENGTH)
-                {
-                    first_block_end = START_BLOCK_LENGTH;
-                }
-
-                auto last_block_end = log.find_last_of('\n', log.size() - END_BLOCK_LENGTH);
-                if (last_block_end == std::string::npos || last_block_end < log.size() - END_BLOCK_MAX_LENGTH)
-                {
-                    last_block_end = log.size() - END_BLOCK_LENGTH;
-                }
-
-                auto first = log.begin() + first_block_end;
-                auto last = log.begin() + last_block_end;
-                auto skipped_lines = std::count(first, last, '\n');
-                log.replace(first, last, fmt::format("\n...\nSkipped {} lines\n...", skipped_lines));
+                first_block_end = start_block_max_length;
             }
 
-            while (!log.empty() && log.back() == '\n')
+            auto last_block_start = log.find_last_of('\n', log.size() - end_block_length);
+            if (last_block_start == std::string::npos || last_block_start < log.size() - end_block_max_length)
             {
-                log.pop_back();
+                last_block_start = log.size() - end_block_max_length;
             }
 
-            return fmt::format("<details><summary>{}</summary>\n\n```\n{}\n```\n</details>", path.native(), log);
+            auto first = log.begin() + first_block_end;
+            auto last = log.begin() + last_block_start;
+            auto skipped_lines = std::count(first, last, '\n');
+            out.append(log, 0, first_block_end);
+            fmt::format_to(std::back_inserter(out), "\n...\nSkipped {} lines\n...", skipped_lines);
+            out.append(log, last_block_start);
         }
-    };
+
+        while (!out.empty() && out.back() == '\n')
+        {
+            out.pop_back();
+        }
+        out += "\n```\n</details>";
+    }
+
+    static void append_logs(const Filesystem& fs, const std::vector<std::string>& logs, int max_size, std::string& out)
+    {
+        int extra_size = 0;
+        max_size -= static_cast<int>(logs.size()) * 2;
+        auto size_per_log = max_size / static_cast<int>(logs.size());
+        for (auto& log : logs)
+        {
+            const auto old_size = out.size();
+            append_log(fs, log, size_per_log + extra_size, out);
+            extra_size += size_per_log - static_cast<int>(out.size() - old_size);
+            out += "\n\n";
+        }
+    }
 
     std::string create_github_issue(const VcpkgCmdArguments& args,
                                     const ExtendedBuildResult& build_result,
                                     const VcpkgPaths& paths,
                                     const InstallPlanAction& action)
     {
+        constexpr int MAX_ISSUE_SIZE = 65536;
         const auto& fs = paths.get_filesystem();
         std::string result;
+        result.reserve(MAX_ISSUE_SIZE);
```

Comment:
This is not a place where perf matters and it isn't at all obvious to me that this is the right reservation even if it were.